scripts/profiler/rtt_nordic_profiler_host.py

```diff
@@ -289,15 +289,37 @@ def _read_single_event_rtt(self):
                 self.after_half = True
 
         timestamp = self._calculate_timestamp_from_clock_ticks(timestamp_raw)
+        data=[]
+        def int32():
+            signum = True
+            buf = self._read_bytes(4)
+            data.append(int.from_bytes(buf, byteorder=self.config['byteorder'],
+                                    signed=signum))
 
-        data = []
-        for i in et.data_types:
+        def uint32():
             signum = False
-            if i[0] == 's':
-                signum = True
             buf = self._read_bytes(4)
             data.append(int.from_bytes(buf, byteorder=self.config['byteorder'],
-                                       signed=signum))
+                                    signed=signum))
+
+        def string_():
+            buf = self._read_bytes(1)
+            buf = self._read_bytes(int.from_bytes(buf, byteorder=self.config['byteorder'],
+                                    signed=False))
+            data.append(buf.decode())
+
+        READ_BYTES = {
```

Comment:
I believe it is better to define this just in place where it is used, since it is just a shortcut to avoid big 'if' ladder. Otherwise it would have to be moved alongside with all functions that are used in this dictionary.